### ISSUE TITLE
rejected promise callbacks are never triggered on saveChanges

### ISSUE THREAD

**Issue by drzaiusx11** [jaystack/jaydata#54]
It seems rejected callbacks are not being triggered properly when a context saveChanges() method is called (and fails).

quick example: say you try to add an item to a database that already exists (key conflict)

``` javascript
var m = new $app.types.TestModel({id: 1, name: "test"});
ctx.TestModel.add(m);
ctx.saveChanges()
.then(
  // resolve callback
  function() { 
    ctx.TestModel.add(m); 
    return ctx.saveChanges();
  },
).then(
  // resolve callback
  function() {} // don't care
  // reject callback
  function() {
    console.log('never gets here, instead throws default error');
  }
).fail(function() {
  console.log('also never gets here, instead throws default error');
});
```

instead of fail() or then()'s rejected callbacks getting called, a "default error" is thrown:

``` javascript
DefaultError: DEFAULT ERROR CALLBACK!
```

This seems like broken usage of the promise API (calling resolve() callbacks, but never reject() callbacks)

**Comment by agdolla** [comment]
Hi
Could you please include more info and source so I can reproduce the problem?
Thanks
Gabor

**Comment by PeterAronZentai** [comment]
We are on it with Victor, and located the error. :) thank you for reporting. Actually we already fixed but needs testing. We are to check it in tomorrow.

In the mean time: the q.js based qDeferred is working with the same syntax.

**Comment by PeterAronZentai** [comment]
The fix has been checked in to development branch as bf9c5ca621288134f4050c3238746380d2a05bec

**Comment by PeterAronZentai** [comment]
The fix is available for download from GIT and will be available as release version 1.2.3
